### PR TITLE
Add group-specific integration key for PagerDuty

### PR DESCRIPTION
**Pull request by achrefbensaadVPaccount**
## What this is doing ?

Provides support for paging multiple teams based on the `group` selector while keeping backward compatibility to the old pagerduty configuration manifest 

-  integration per team can be specified in the integrations sections in an array form.

```yaml
alerting:
  pagerduty: 
    integration-key: "00000000000000000000000000000000"
    integrations:
      - integration-key: "00000000000000000000000000000000"
        group: "team"
```

behavior:
- team integration have priority over the general integration (old pagerduty configuration)
- if no team integration is provided it will defaults to the general pagerduty integration 
- if no team integration and no general integration were provided it defaults to the first team integration provided

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -414,18 +414,32 @@ services:
 | Parameter                                | Description                                                                   | Default        |
 |:---------------------------------------- |:----------------------------------------------------------------------------- |:-------------- |
 | `alerting.pagerduty`                     | Configuration for alerts of type `pagerduty`                                  | `{}`           |
-| `alerting.pagerduty.integration-key`     | PagerDuty Events API v2 integration key.                                      | Required `""`  |
+| `alerting.pagerduty.integration-key`     | PagerDuty Events API v2 integration key.                                      | `""`           |
 | `alerting.pagerduty.default-alert`       | Default alert configuration. <br />See [Setting a default alert](#setting-a-default-alert) | N/A       |
+| `alerting.pagerduty.integrations`        | Pagerduty integrations per team configurations                                |                |
+| `alerting.pagerduty.integrations`        | Pagerduty integrations per team configurations                                |   `[]`         |
```
Comment (TwinProduction) [inline README.md:420]:
```suggestion
```
Could you remove these two duplicate lines?

File: README.md [hunk 2]
```diff
@@ -414,18 +414,32 @@ services:
...
+- If no team integration and no general integration were provided it defaults to the first team integration provided
+
+
 ```yaml
 alerting:
   pagerduty: 
     integration-key: "********************************"
+    intergrations:
+     - integration-key: "********************************"
+       group: "core"
```
Comment (TwinProduction) [inline README.md:442]: So if I understand correctly, `alerting.pagerduty.integration-key` is the fallback integration key and the `alerting.pagerduty.integrations[]` with a matching group name takes precedence over the fallback integration key, right?

File: alerting/provider/pagerduty/pagerduty.go [hunk 3]
```diff
@@ -13,22 +13,55 @@ const (
...
 // IsValid returns whether the provider's configuration is valid
 func (provider *AlertProvider) IsValid() bool {
-	return len(provider.IntegrationKey) == 32
+	groups := make(map[string]int)
+	if provider.Integrations != nil {
+		for _, integration := range provider.Integrations {
+			if _, present := groups[integration.Group]; present || integration.Group == "" || len(integration.IntegrationKey) != 32 {
+				return false
+			}
+			groups[integration.Group] = 1
```
Comment (TwinProduction) [inline alerting/provider/pagerduty/pagerduty.go:39]:
```suggestion
	registeredGroups := make(map[string]bool)
	if provider.Integrations != nil {
		for _, integration := range provider.Integrations {
			if isAlreadyRegistered := registeredGroups[integration.Group]; isAlreadyRegistered || integration.Group == "" || len(integration.IntegrationKey) != 32 {
				return false
			}
			registeredGroups[integration.Group] = true
```

For the sake of making this a little easier to read

File: alerting/provider/pagerduty/pagerduty.go [hunk 4]
```diff
@@ -13,22 +13,55 @@ const (
...
+			if group == itegration.Group {
+				return itegration.IntegrationKey
+			}
+		}
+	}
+	if provider.IntegrationKey != "" {
+		return provider.IntegrationKey
+	}
+	return provider.Integrations[0].IntegrationKey
 }
```
Comment (TwinProduction) [inline alerting/provider/pagerduty/pagerduty.go:59]:
```suggestion
// GetPagerDutyIntegrationKey returns the appropriate pagerduty integration key
func (provider *AlertProvider) GetPagerDutyIntegrationKey(group string) string {
	if provider.Integrations != nil {
		for _, integration := range provider.Integrations {
			if group == integration.Group {
				return integration.IntegrationKey
			}
		}
	}
	if provider.IntegrationKey != "" {
		return provider.IntegrationKey
	}
	return ""
}
```

**Review by TwinProduction — CHANGES_REQUESTED** [review]

File: README.md (hunk 2 above)
Comment (achrefbensaadVPaccount) [reply README.md:440]: Yes Correct.

File: README.md (hunk 1 above)
Comment (achrefbensaadVPaccount) [reply README.md]: Sure

**Review by TwinProduction — APPROVED** [review]

**Comment by TwinProduction** [comment]
@achrefbensaadVPaccount I made some adjustments in a6bc0039e986cd79af21c2ef657f5eef31bf4a10 to support future use cases.

Namely, `integrations` has been renamed to `overrides`.

See https://github.com/TwinProduction/gatus#configuring-pagerduty-alerts

**Comment by TwiN** [comment]
Released in [v3.2.2](https://github.com/TwiN/gatus/releases/tag/v3.2.2)

**Pull request merged**